llvm/lib/Object/MachOObjectFile.cpp

```diff
@@ -118,7 +104,10 @@ static const char *getPtr(const MachOObjectFile &O, size_t Offset,
 static MachO::nlist_base
 getSymbolTableEntryBase(const MachOObjectFile &O, DataRefImpl DRI) {
   const char *P = reinterpret_cast<const char *>(DRI.p);
-  return getStruct<MachO::nlist_base>(O, P);
+  auto NListOrErr = getStructOrErr<MachO::nlist_base>(O, P);
+  if (!NListOrErr)
+    report_fatal_error(NListOrErr.takeError());
```

Comment:
Please add a FIXME here to indicate that this is still not ideal: the error should be propagated up the stack, to be reported using the tool's error reporting mechanism.

The same comment applies to every other place you've changed where there's a `report_fatal_error` call now.